docusaurus/docs/recipes/contentful.md

```diff
@@ -0,0 +1,514 @@
+            }
+        }
+    ],
+}
+```
+
+- Add a SEO Assets field (many file, media input) to the model.
+- Create a link field type in the models you need SEO and link it to the model you just created.
+
+The resulting content model should look like this.
+
+![SEO Content Model](../../static/img/SEO%20content%20model.png)
+
+In order to obtain the `id` used for meta tags whose content is an asset, you need to select the asset entry and, on the right side panel of the asset, select **Info** and copy the **id**. Later in the application you'll use this **id** to generate the URL for the asset.
+
+On the application itself, the steps to customize the SEO are the following:
+
+- Define a _default_ SEO data.
+- Fetch page data as you normally would.
+- Obtain the SEO data from the Contentful page data (example code for the Contentful SEO _parser_)
+
+```js
+import { mapValues } from 'lodash';
+
+// This function will determine if a meta entry's content is an id
+// If it is, it obtains the url to the asset that matches the id
+// Otherwise it returns the content 
+
+const getContent = (content, assets) => {
+    if (isPlainObject(content)) {
+        const asset = assets.find(({ sys }) => sys.id === content.id);
+
+        return asset.fields.file.url.slice(2);
+    }
+
+    return content;
+};
+
+// Provided with the result from the Contentful API call and the entryID for the current page content,
+// This function will extract the SEO data, and complete it with the URL of any needed asset.
+
+const contentfulSEOParser = (contentfulData, entryID) => {
+    const entry = contentfulData.items.find((item) => item.sys.id === entryID);
+    const seoID = entry.fields.seo.id;
+    const seoEntry = entry.includes.Entry.find((entry) => entry.sys.id === seoID);
+    let seoContent;
+
+    try {
+        seoContent = JSON.parse(seoEntry.fields.seo);
+    } catch (parsingError) {
+        console.err('Unvalid JSON in SEO content');
+        console.err(parsingError);
+        return;
+    }
+
+    // Convert every content: { id: ''} into a content: url, keep unchanged otherwise
+    const content = {
+        title: seoContent.title,
+        meta: seoContent.meta.map((entry) => mapValues(entry, (value) => getContent(value, contentfulData.includes.Asset)))
+        link: seoContent.link.map((entry) => mapValues(entry, (value) => getContent(value, contentfulData.includes.Asset)))
+    }
+
+    return content;
+}
+```
+
+- Merge the _default_ SEO data with the SEO data fetched from the Contentful.
+
+```js
+const seoData = {
+    title: contentfulSEO.title || defaultSEO.title,
+    meta: [...defaultSEO.meta, ...contentfulSEO.meta],
+    link: [...defaultSEO.link, ...contentfulSEO.link],
+}
+```
+
+- Render the SEO data (title and tags) inside the `Head` in the `App.js` file.
+
+For the rendering step, we recommend using [@moxy/next-seo](https://www.npmjs.com/package/@moxy/next-seo), as this package takes care of two things:
```

Comment:
I don't think this should be only a recommendation as we already use the package in the boilerplate, I would enforce the use of `next-seo`. Also the point of creating the parser is exactly to transform the information to be used in the `Seo` component. Thoughts?

Reply:
Makes sense, we can change the wording from "recommendation" to "use `next-seo`".